### ISSUE TITLE
No such file or directory: '/sys/class/power_supply'

### ISSUE THREAD

**Issue by SailorMax** [nicolargo/glances#1063]
#### Description
after upgrade 2.8.2 -> 2.8.8 I can't start glances anymore:
```
Traceback (most recent call last):
  File "/usr/local/bin/glances", line 9, in <module>
    load_entry_point('Glances==2.8.8', 'console_scripts', 'glances')()
  File "/usr/local/lib/python2.7/dist-packages/glances/__init__.py", line 225, in main
    start_standalone(config=config, args=args)
  File "/usr/local/lib/python2.7/dist-packages/glances/__init__.py", line 105, in start_standalone
    standalone = GlancesStandalone(config=config, args=args)
  File "/usr/local/lib/python2.7/dist-packages/glances/standalone.py", line 43, in __init__
    self.stats = GlancesStats(config=config, args=args)
  File "/usr/local/lib/python2.7/dist-packages/glances/stats.py", line 43, in __init__
    self.load_modules(self.args)
  File "/usr/local/lib/python2.7/dist-packages/glances/stats.py", line 77, in load_modules
    self.load_plugins(args=args)
  File "/usr/local/lib/python2.7/dist-packages/glances/stats.py", line 105, in load_plugins
    self._plugins[plugin_name] = plugin.Plugin(args=args)
  File "/usr/local/lib/python2.7/dist-packages/glances/plugins/glances_sensors.py", line 61, in __init__
    self.batpercent_plugin = BatPercentPlugin(args=args)
  File "/usr/local/lib/python2.7/dist-packages/glances/plugins/glances_batpercent.py", line 56, in __init__
    self.glancesgrabbat = GlancesGrabBat()
  File "/usr/local/lib/python2.7/dist-packages/glances/plugins/glances_batpercent.py", line 98, in __init__
    self.bat = batinfo.batteries()
  File "/usr/local/lib/python2.7/dist-packages/batinfo/battery.py", line 105, in __init__
    self.update()
  File "/usr/local/lib/python2.7/dist-packages/batinfo/battery.py", line 114, in update
    for dirname in os.listdir(self.bat_root_path):
OSError: [Errno 2] No such file or directory: '/sys/class/power_supply'
```

This is dedicated server. It has no battery.

#### Versions

* Glances (glances -V): 2.8.8
* PSutil (glances -V): 5.2.1
* Operating System (lsb_release -a): Debian 7.11

#### Logs
```
2017-03-25 20:08:14,671 -- INFO -- Start Glances 2.8.8
2017-03-25 20:08:14,671 -- INFO -- CPython 2.7.3 and PSutil 5.2.1 detected
2017-03-25 20:08:14,678 -- INFO -- Read configuration file '/etc/glances/glances.conf'
2017-03-25 20:08:14,679 -- INFO -- Start standalone mode
2017-03-25 20:08:14,689 -- DEBUG -- Stats history activated for plugin memswap (items: ['percent'])
2017-03-25 20:08:14,738 -- DEBUG -- cloud plugin - Create thread for AWS EC2
2017-03-25 20:08:14,804 -- DEBUG -- No [ports] section in the configuration file. Cannot load ports list.
2017-03-25 20:08:14,805 -- DEBUG -- Stats history activated for plugin load (items: ['min1', 'min5', 'min15'])
2017-03-25 20:08:14,806 -- DEBUG -- Stats history activated for plugin mem (items: ['percent'])
2017-03-25 20:08:14,807 -- DEBUG -- Stats history activated for plugin diskio (items: ['read_bytes', 'write_bytes'])
2017-03-25 20:08:15,126 -- DEBUG -- IP plugin - Cannot open URL https://api.ipify.org/?format=json (<urlopen error [Errno 1] _ssl.c:504: error:14077438:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert internal error>)
2017-03-25 20:08:17,756 -- DEBUG -- cloud plugin - Cannot connect to the AWS EC2 API http://169.254.169.254/latest/meta-data/placement/availability-zone: HTTPConnectionPool(host='169.254.169.254', port=80): Max retries exceeded with url: /
```

**Comment by nicolargo** [comment]
Try to uninstall batinfo. It should solve the issue.

**Comment by SailorMax** [comment]
@nicolargo 
Uninstall batinfo helped. Now work as expected.
Can you output some warning instead of crash? Or make workaround?

**Comment by nicolargo** [comment]
Should be corrected in the future 2.9 version.
